default.nix

```diff
@@ -206,12 +206,29 @@ let
     load-config-from-site = writeTextDir "share/emacs/site-lisp/default.el" ''
       (message "doom-emacs is not placed in `doom-private-dir',
       loading from `site-lisp'")
-      (when (> emacs-major-version 26)
-            (load "${doom-emacs}/early-init.el"))
+      ${lib.optionalString (!isEmacs29) ''
+      (load "${doom-emacs}/early-init.el")
       (load "${doom-emacs}/core/core-start.el")
+      ''}
     '';
   in (emacsPackages.emacsWithPackages (epkgs: [ load-config-from-site ]));
 
+  # create a `emacs.d` dir to be loaded using `--init-directory` flag from Emacs 29+
+  # this will allow proper usage of `early-init.el`, fixing FOUC issues and improving
+  # startup performance
+  init-el = pkgs.writeTextDir "emacs.d/init.el" ''
+    (load "${doom-emacs}/core/core-start.el")
+  '';
+
+  early-init-el = pkgs.writeTextDir "emacs.d/early-init.el" ''
+    (load "${doom-emacs}/early-init.el")
+  '';
```

Comment:
This is a no-no. Perhaps move it to `share/doom.d` or something like that.

Reply:
Any reason why? Those are `emacs.d` files, they're basically files that you would clone to `~/.emacs.d` in a normal doom-emacs installation. And I don't really understand the point of putting them under a `share` directory either.

Reply:
@thiagokokada It makes it impossible to use tools like [Chemacs](https://github.com/plexus/chemacs2)

Reply:
@nwjsmith Not sure I am understand why though. Like I said in other PR, you can pass `--init-directory` that will ignore the incantation in the wrapper. Also, this is not stored in any way that is accessible for Emacs (the only way that Emacs now that this directory exists is because of the wrapper).

Reply:
It's not *needed*, but it's currently pretty confusing, because instead of either a `share` (implies this will be exposed to `systemPackages`/etc) or just having it in the store path's [relative] root directory. (it's probably private) 

In this case, it's in neither which throws that metric off and forces any new readers of the code to see where it's referenced a bit less lazily.

Reply:
Done.